docs/readme/error_handling.md

```diff
@@ -0,0 +1,22 @@
+# Error Handling
+
+In SITUP Pipelines Errors should be handled by implementation of plugin components and should not throw any uncaught runtime exceptions. If thrown the pipeline will stop immediately or halt eventually based on the component that throws the exception.
+
+Below are the different scenario of error,
+
+## Single Pipeline
+Single pipeline is a pipeline that doesn't use the pipeline connectors.
+
+* If an exception encountered when creating a pipeline, the pipeline stops and exits using `system.exit(1)`. 
+* If an exception encountered during pipeline `start()` method, the pipeline stops with appropriate logs. 
+* If an exception encountered in Pipeline `processors` or `sinks` during runtime, the pipeline stops with appropriate logs.
+
+## Connected Pipelines
+
+Connected Pipelines is scenario where two or more pipelines connected using the Pipeline Connector.
+
+
+* If an exception encountered when creating any of the connected pipelines, all the connected pipelines stop and exits using `system.exit(1)`. 
+* If an exception encountered during pipeline `start()` method in any of the connected pipelines, the pipeline that encountered the exception will stop and other pipelines will run and will not receive or process any data as the one of the connected pipelines is unavailable.
```

Comment:
say source fails to start -> this would result in sink not to start (in a PipelineConnector scenario), and it will eventually shutdown everything;

Reply:
Are you sure, I referred this [issue](https://github.com/opendistro-for-elasticsearch/simple-ingest-transformation-utility-pipeline/issues/90#issue-722490676), and it says it will halt not shutdown.

Reply:
Yes, in that issue - I broke it into 2 cases right -> if root source fails to start, the other pipelines will receive nothing; the second case is any of the connected pipeline's source fails to start (_this technically is very rare because any of the pipeline source (other than root) is definitely PipelineConnector and its start method is just an assignment operation_) but if it happens - pipeline will shutdown right because it is effectively sink failed for the connected pipeline right

Reply:
Exactly, I ignored the PipelineConnector::Start from the docs because that is never going to happen.